packages/tentales/src/middleware/create-service-middlewares.ts

```diff
@@ -0,0 +1,54 @@
+import { createLog } from "tentales-log"
+import { renderer } from "tentales-renderer"
+import { data } from "tentales-data"
+import { editor } from "tentales-editor"
+import { convertServiceMethodsToServices } from "../service/utils"
+import { uppercaseFirst } from "../utils/string"
+import {
+  Middleware,
+  MiddlewareDefinition,
+  ServiceCaller,
+  TenTalesService,
+} from "../types"
+
+const SERVICE_MODULES: {
+  renderer: TenTalesService
+  data: TenTalesService
+  editor: TenTalesService
+  [key: string]: TenTalesService
```

Comment:
This is needed to be able to do `SERVICE_MODULES[serviceMethod.name]` on line 34. But it would be nice to limit the key to the properties available on the object. Anyone knows if that can be done?

Reply:
afaik it's not :/
```ts
type PossibleKeys = "one" | "two" | "three"
...
[key: PossibleKeys]: TenTalesService
```
would have been nice, but as far as I know not possible. It must be `string` I think, when dynamic.

Reply:
Maybe I am missing something here, but isn't this exactly what an enum is for? 🤔

Reply:
Yes... In the best of worlds :) 

But here the thing is that Typescript doesn't allow anything but `string` type as a dynamically typed key of an object. But a union type or even better an enum would have have been better, agree!

Reply:
Hmm... look a bit at this. It is possible with `mapped types`:

```ts
enum Keys { 
  ONE = "one",
  TWO = "two",
  THREE = "three"
}

type Config = {
  [K in Keys]: String
}
```
or 
```ts
type Keys = "one" |  "two" | "three"

type Config = {
  [K in Keys]: String
}
```

So yeah :) probably a enum here for possible serives modules could be nice :)

Reply:
Awesome, cheers @janneh!